compiler/shared/test/diff/mono.mls

```diff
@@ -0,0 +1,1259 @@
+//│ res
+//│     = 1
+//│ res
+//│     = 3
+
+:mono
+class Cons(e: Int, tail: Cons | Nil) {
+  fun count(): Int
+  fun count() = 1 + tail.count()
+}
+class Nil() {
+  fun count() = 0
+}
+fun foo(x) = 
+  x(new Cons(1, new Nil())) + x(new Nil())
+foo(l => l.count())
+foo(l => (new Cons(2, l)).count())
+//│ Parsed:
+//│ TypingUnit(NuTypeDef(class, Cons, (), Tup(e: Var(Int), tail: App(Var(|), Tup(_: Var(Cons), _: Var(Nil)))), (), None, None, TypingUnit(NuFunDef(None, count, None, [], PolyType(List(),Function(Tuple(List()),TypeName(Int)))), NuFunDef(None, count, None, [], Lam(Tup(), App(Var(+), Tup(_: IntLit(1), _: App(Sel(Var(tail), count), Tup()))))))), NuTypeDef(class, Nil, (), Tup(), (), None, None, TypingUnit(NuFunDef(None, count, None, [], Lam(Tup(), IntLit(0))))), NuFunDef(None, foo, None, [], Lam(Tup(_: Var(x)), Blk(...))), App(Var(foo), Tup(_: Lam(Tup(_: Var(l)), App(Sel(Var(l), count), Tup())))), App(Var(foo), Tup(_: Lam(Tup(_: Var(l)), App(Sel(Bra(rcd = false, New(Some((TypeName(Cons),[2, l,])), TypingUnit())), count), Tup())))))
+//│ Lifted:
+//│ TypingUnit {
+//│   class Cons$1([e: Int, tail: |(Cons, Nil,),]) {
+//│     fun count = () -> Int
+//│     fun count = () => +(1, ((this).tail).count(),)
+//│   }
+//│   class Nil$2([]) {fun count = () => 0}
+//│   class Lambda1$2$3([]) {fun apply = (l,) => (l).count()}
+//│   class Lambda1$3$4([]) {
+//│     fun apply = (l,) => ('(' new Cons$1([2, l,]) {} ')').count()
+//│   }
+//│   fun foo$1 = (x,) => {+(x(new Cons$1([1, new Nil$2([]) {},]) {},), x(new Nil$2([]) {},),)}
+//│   Code(List(foo$1({new Lambda1$2$3([]) {}},)))
+//│   Code(List(foo$1({new Lambda1$3$4([]) {}},)))
+//│ }
+//│ Mono:
+//│ 
+//│ Defunc result: 
+//│ main$$5()
+//│ main$$6()
+//│ fun count$Cons$1(this) =
+//│   +(#1, this.tail match {case obj: Cons$1 => count$Cons$1(obj); case obj: Nil$2 => count$Nil$2(obj)})
+//│ fun foo$1(x) =
+//│   +(x match {case obj: Lambda1$3$4 => apply$Lambda1$3$4(obj, new Cons$1 (#1, new Nil$2 () ) ); case obj: Lambda1$2$3 => apply$Lambda1$2$3(obj, new Cons$1 (#1, new Nil$2 () ) )}, x match {case obj: Lambda1$3$4 => apply$Lambda1$3$4(obj, new Nil$2 () ); case obj: Lambda1$2$3 => apply$Lambda1$2$3(obj, new Nil$2 () )})
+//│ fun count$Nil$2(this) =
+//│   #0
+//│ fun main$$6() =
+//│   foo$1(new Lambda1$3$4 () )
+//│ fun main$$5() =
+//│   foo$1(new Lambda1$2$3 () )
+//│ fun apply$Lambda1$3$4(this, l) =
+//│   new Cons$1 (#2, l)  match {case obj: Cons$1 => count$Cons$1(obj)}
+//│ fun apply$Lambda1$2$3(this, l) =
+//│   l match {case obj: Cons$1 => count$Cons$1(obj); case obj: Nil$2 => count$Nil$2(obj)}
+//│ class Lambda1$3$4() {
+//│ }
+//│ class Nil$2() {
+//│ }
+//│ class Cons$1(e, tail) {
+//│ }
+//│ class Lambda1$2$3() {
+//│ }
+//│ class Cons(e: Int, tail: Cons | Nil) {
+//│   fun count: () -> Int
+//│ }
+//│ class Nil() {
+//│   fun count: () -> 0
+//│ }
+//│ fun foo: ((Cons | Nil) -> Int) -> Int
+//│ Int
+//│ res
+//│     = 1
+//│ res
+//│     = 3
+
+:mono
+:e // FIXME: Malformed Test?
```

Comment:
@yuankeyu could you comment on this and propose a fixa? What were you trying to do?